### ISSUE TITLE
Expectations Breakpoint issue

### ISSUE THREAD

**Issue by bishalg** [airbnb/lottie-ios#959]
## Check these before submitting:
- [x] The issue doesn't involve an [Unsupported Feature](https://github.com/airbnb/lottie-ios/blob/master/README.md#supported-after-effects-features)
- [x] This issue isn't related to another open issue

## This issue is a:
- [x] Non-Crashing Bug (Visual or otherwise)

## Which Version of Lottie are you using?
Lottie 3.0

## What Platform are you on? 
- [x] iOS

## What Language are you in?
- [x] Swift

## Expected Behavior 
 - The application should not stop due to debug breakpoint 

## Actual Behavior
<img width="958" alt="Exceptatino Breakpoint" src="https://user-images.githubusercontent.com/805689/59961311-a94ffa00-94f5-11e9-8203-65df8637646c.png">

Application has enabled Debug breakpoint and there is Lottie animation on the start screen, hence the application will hit this exception breakpoint on each run.

## Code Example
```
public init(from decoder: Decoder) throws {
    var container = try decoder.unkeyedContainer()
    
    if !container.isAtEnd {
      self.x = try container.decode(Double.self)
    } else {
      self.x = 0
    }
    
}

```
## Animation JSON
https://lottiefiles.com/3015-man-and-phone

**Comment by bishalg** [comment]
Another Exception Breakpoint issue - 

**Screenshot**
<img width="1009" alt="Keyframe group" src="https://user-images.githubusercontent.com/805689/59965642-ab837a00-9530-11e9-80aa-d632e080d036.png">

**Code Example** 
```
required init(from decoder: Decoder) throws {
    let container = try decoder.container(keyedBy: KeyframeWrapperKey.self)
    
    if let keyframeData: T = try? container.decode(T.self, forKey: .keyframeData) {
      /// Try to decode raw value; No keyframe data.
      self.keyframes = [Keyframe<T>(keyframeData)]
    } else 
```

**File Name** 
KeyframeGroup.swift

**Comment by buba447** [comment]
This is well known and unfortunately theres really no work around. This is how the codable container handles optional values...

**Comment by JoseG95** [comment]
I have the same issue with the KeyframeGroup. How can I solve it?

**Comment by oblq** [comment]
Have you tried decodeIfPresent()?
try? container. decodeIfPresent(T.self, forKey: .key)

This will not hit the Swift error breakpoint.

**Comment by jalone87** [comment]
Please consider this issue with a bit more priority, dynamic apps with heavy lottie usage at startup make it very annoying to run with a debugger attached.

**Comment by qazwyw** [comment]
Is this issus resolved?

**Comment by basheerpaliyathu** [comment]
stll not resolved

**Comment by calda** [comment]
You can probably work around this by using the new dictionary decoding implementation available in Lottie 3.4.0 (which doesn't use `Decodable`).
